include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,123 @@
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+
+        /// Distributes `n >= 2` values linearly in the closed interval [`from`, `to`].
+        template<typename T>
+        struct linear_distribute_view
+          : view_facade<linear_distribute_view<T>, finite>
+        {
+            static_assert(std::is_arithmetic<T>(),
+                          "T must model the Arithmetic<T> concept; that is, "
+                          "it must be one of the native integer or floating-point types");
+        private:
+            friend range_access;
+
+            T from_, to_, spacing_;
+
+            constexpr T read() const
+            {
+                return from_;
+            }
+            constexpr void next()
+            {
+                from_ += spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_integral<T>())
+            constexpr bool equal(default_sentinel) const
+            {
+                return from_ == to_ + spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_floating_point<T>())
+            constexpr bool equal(default_sentinel) const
+            {
+                return from_ > to_;
+            }
+            constexpr void prev()
+            {
+                from_ -= spacing_;
+            }
+            constexpr void advance(std::ptrdiff_t n)
+            {
+                RANGES_EXPECT(to_ - from_ >= n * spacing_);
+                from_ += n * spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_integral<T>())
+            constexpr std::ptrdiff_t distance_to(linear_distribute_view const &that) const
+            {
+                return (that.from_ - from_) / spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_floating_point<T>())
+            constexpr std::ptrdiff_t distance_to(linear_distribute_view const &that) const
+            {
+                return static_cast<std::ptrdiff_t>(std::floor((that.from_ - from_) / spacing_));
+            }
+        public:
+            constexpr linear_distribute_view() = default;
+            constexpr linear_distribute_view(T from, T to, std::size_t n)
+              : from_(detail::move(from)), to_(detail::move(to))
+            {
+              RANGES_EXPECT(n > 1);
+              RANGES_EXPECT(to_ > from_);
+              spacing_ = (to_ - from_)/ static_cast<T>(n - 1);
+              RANGES_EXPECT(spacing_ > T{0});
+            }
+        };
+
+        ///@}
+
+        namespace view
+        {
+            /// Distributes `n >= 2` values linearly in the closed interval `[from, to]`.
+            struct linear_distribute_fn
+            {
+                template<typename T,
+                    CONCEPT_REQUIRES_(std::is_arithmetic<T>())>
+                constexpr auto operator()(T from, T to, std::size_t n) const
+                    RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT(linear_distribute_view<T>{from, to, n})
```

Comment:
Typical style is to not indent `RANGES_DECLTYPE_AUTO_XXX`, and to put the parens where braces would go in a "real" function body:
```c++
constexpr auto operator/*...*/
RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
(
    linear_distribute_view<T>{from, to, n}
)
```